(dp1
S'output'
p2
S"<type 'exceptions.IndexError'> list index out of range"
p3
sS'layer'
p4
S'/home/marco/trabajo/experimental/web2py/applications/sisapretty/controllers/consultas.py'
p5
sS'code'
p6
S'# -*- coding: utf-8 -*-\n\n@auth.requires_login()\ndef certificados():\n\n    cliente=db(db.clientes.id==request.args(1)).select()\n    lugar=db(db.establecimientos.id==request.args(0)).select()\n    response.subtitle+=" Certificados del cliente "+cliente[0].nombre.capitalize()+" "+cliente[0].apellido.capitalize()+" /  "+lugar[0].nombre\n    certificados=db(db.certificados.establecimiento==request.args(0)).select()\n\n    db.certificados.establecimiento.default=request.args(0)\n    db.certificados.establecimiento.writable=False\n    db.certificados.establecimiento.readable=False\n    nCertificado=SQLFORM(db.certificados)\n\n    if nCertificado.process().accepted:\n        db.cobros.insert(certificado=nCertificado.vars.id, importe=0.0, saldo=0.0)\n        redirect(URL(\'consultas\', \'certificados\', args=(request.args(0), request.args(1))))\n        session.flash=\'El certificado ha sido agregado\'\n        \n    elif nCertificado.errors:\n        response.flash=\'El certificado tiene errores\'\n        \n    return dict(certificados=certificados, nCertificado=nCertificado)\n\n@auth.requires_login()\ndef cobros():\n    cliente=db(db.clientes.id==request.args(0)).select()\n    response.subtitle+=" Cobros al cliente "+cliente[0].nombre.capitalize()+" "+cliente[0].apellido.capitalize()\n    cobros=db((db.certificados.establecimiento==db.establecimientos.id)&(db.cobros.certificado==db.certificados.id)&(db.establecimientos.cliente==request.args(0))).select()\n\n    return dict(cobros=cobros)\n    \n@auth.requires_login()\ndef establecimientos():\n\n    cliente=db(db.clientes.id==request.args(0)).select()\n    response.subtitle+=" Establecimientos del cliente "+cliente[0].nombre.capitalize()+" "+cliente[0].apellido.capitalize()\n\n    db.establecimientos.cliente.default=request.args(0)\n    db.establecimientos.cliente.writable=False\n    db.establecimientos.cliente.readable=False\n    \n    sucursales=db(db.establecimientos.cliente==request.args(0)).select()\n    \n    nSucursal=SQLFORM(db.establecimientos)\n\n    if nSucursal.process().accepted:\n        redirect(URL(\'consultas\',\'establecimientos\', args=(request.args(0))))\n    elif nSucursal.errors:\n        response.flash="Revise los datos ingresados"\n    \n    return dict(nSucursal=nSucursal, sucursales=sucursales)\n\n@auth.requires_login()\ndef imprimircertificado():\n    certificado=db((db.certificados.id==request.args(0))&(db.certificados.establecimiento==db.establecimientos.id)&(db.clientes.id==db.establecimientos.cliente)).select()\n    return dict(certificado=certificado)\n\nresponse._vars=response._caller(certificados)\n'
p7
sS'snapshot'
p8
(dp9
sS'traceback'
p10
S'Traceback (most recent call last):\n  File "/home/marco/trabajo/experimental/web2py/gluon/restricted.py", line 220, in restricted\n    exec ccode in environment\n  File "/home/marco/trabajo/experimental/web2py/applications/sisapretty/controllers/consultas.py", line 60, in <module>\n  File "/home/marco/trabajo/experimental/web2py/gluon/globals.py", line 389, in <lambda>\n    self._caller = lambda f: f()\n  File "/home/marco/trabajo/experimental/web2py/gluon/tools.py", line 3337, in f\n    return action(*a, **b)\n  File "/home/marco/trabajo/experimental/web2py/applications/sisapretty/controllers/consultas.py", line 8, in certificados\n    response.subtitle+=" Certificados del cliente "+cliente[0].nombre.capitalize()+" "+cliente[0].apellido.capitalize()+" /  "+lugar[0].nombre\n  File "/home/marco/trabajo/experimental/web2py/gluon/dal.py", line 10959, in __getitem__\n    row = self.records[i]\nIndexError: list index out of range\n'
p11
s.